### PR TITLE
Support iam policy member of domain type in iam scanner's required mode,

### PR DESCRIPTION
**Pull request by arbeit**
Hello there!

This fixes issue #799. Please take a look and let me know what you think. Thanks!

### CODE REVIEW

File: google/cloud/security/common/gcp_type/iam_policy.py [hunk 1]
```diff
@@ -288,6 +288,28 @@ def create_from(cls, member):
             member_name = identity_parts[1]
         return cls(identity_parts[0], member_name=member_name)
 
+    def _user_is_in_domain(self, other):
+        """Determine whether user belongs to domain.
```
Comment (blueandgold) [inline google/cloud/security/common/gcp_type/iam_policy.py]: Technically, not a ```user```.  But more clear as ```IAM policy member```.  So:
```
"""Determine whether IAM policy member belongs to domain.
```

File: google/cloud/security/common/gcp_type/iam_policy.py [hunk 2]
```diff
@@ -288,6 +288,28 @@ def create_from(cls, member):
...
+
+        This applies to a situation where a rule has a `domain` style `members`
+        specification and the policy to check specifies users.
+
+        Args:
+            other (IamPolicyMember): The policy binding to check.
```
Comment (blueandgold) [inline google/cloud/security/common/gcp_type/iam_policy.py]: Please clarify this, by adding ```member``` like:
```
other (IamPolicyMember): The policy binding member to check.
```

File: tests/scanner/audit/iam_rules_engine_test.py [hunk 3]
```diff
@@ -1405,6 +1403,87 @@ def test_project_required(self):
         ])
         self.assertItemsEqual(expected_violations, actual_violations)
 
+    def test_policy_all_projects_must_have_owners_from_domain(self):
```
Comment (blueandgold) [inline tests/scanner/audit/iam_rules_engine_test.py]: Please clarify this method name, to show that it's using the domain member type.  So, a good name would be:
```
test_policy_all_projects_must_have_owners_from_domain_type
```

File: tests/scanner/audit/iam_rules_engine_test.py [hunk 4]
```diff
@@ -1405,6 +1403,87 @@ def test_project_required(self):
...
+
+        actual_violations = set(rules_engine.find_policy_violations(
+            self.project1, policy))
+
+        # expected
+        expected_violations = set()
+
+        self.assertEqual(expected_violations, actual_violations)
+
+    def test_policy_all_projects_must_have_owners_from_wildcard_domain(self):
```
Comment (blueandgold) [inline tests/scanner/audit/iam_rules_engine_test.py]: Please clarify this method name, to show that it's using the wildcard domain from user type.  So, a good name would be:
```
test_policy_all_projects_must_have_owners_from_wildcard_domain_of_user_type
```

File: google/cloud/security/common/gcp_type/iam_policy.py [hunk 5]
```diff
@@ -288,6 +288,28 @@ def create_from(cls, member):
...
+
+        Returns:
+            bool: True if `other` is a member of the domain, False otherwise.
+        """
+        if self.type != 'domain' or other.type != 'user':
+            return False
+
+        try:
+            _, domain = other.name.rsplit('@', 1)
```
[the remark this answers is not in the document]
Comment (arbeit) [reply google/cloud/security/common/gcp_type/iam_policy.py]: It's a Python idiom for "I do not care about that value"

**Comment by arbeit** [comment]
@blueandgold -- addressed your review comments, please take a look. Thanks!

**Comment by arbeit** [comment]
@blueandgold not sure about whitelist / blacklist mode .. maybe we can open a separate issue for that?

**Comment by arbeit** [comment]
@blueandgold once this PR lands, I will port the changes to `2.0-dev`

**Pull request merged**